assessments/projects/coredns/self-assessment.md

```diff
@@ -0,0 +1,267 @@
+# Self-assessment
+Authors: Tom Zhang, Rohit Chaudhari, Maryam Mohagheghi, Vamshi Madineni
+
+Contributors/Reviewers: Pranava Kumar Vemula (@Rana-KV).
+
+This document evaluates the design goals for [CoreDNS](https://coredns.io/) as provides a security assessment of the software.
+
+# Self-assessment outline
+
+## Table of contents
+
+* [Metadata](#metadata)
+  * [Security links](#security-links)
+* [Overview](#overview)
+  * [Actors](#actors)
+  * [Actions](#actions)
+  * [Background](#background)
+  * [Goals](#goals)
+  * [Non-goals](#non-goals)
+* [Self-assessment use](#self-assessment-use)
+* [Security functions and features](#security-functions-and-features)
+* [Project compliance](#project-compliance)
+* [Secure development practices](#secure-development-practices)
+* [Security issue resolution](#security-issue-resolution)
+* [Appendix](#appendix)
+
+## Metadata
+
+A table at the top for quick reference information, later used for indexing.
+
+|   |  |
+| -- | -- |
+| Software | [CoreDNS](https://github.com/coredns/coredns) |
+| Security Provider | No |
+| Languages | Golang |
+| | |
+
+### Security links
+
+| Doc | url |
+| -- | -- |
+| Core53 | [Audit Report](https://coredns.io/assets/DNS-01-report.pdf) |
+| Trail of Bits | [Audit Report](https://github.com/trailofbits/publications/blob/master/reviews/CoreDNS.pdf) |
+
+## Overview
+
+CoreDNS is a fast and flexible DNS server/forwarder, written in Go, that chains plugins to manage your DNS data and provide out-of-box functionality.
+
+### Background
+
+CoreDNS is a powerful and efficient DNS server/forwarder, written in Go, specifically designed for modern, dynamic, and containerized environments. Its plugin-based architecture provides flexibility and extensibility, enabling it to accommodate various communication protocols, including TCP, UDP, TLS, HTTP/2, QUIC, and gRPC.
+
+CoreDNS was developed to address the limitations of traditional DNS solutions in cloud-native architectures, with a focus on simplicity, efficiency, and maintainability. The choice of the Go programming language reflects a commitment to balancing performance and readability. CoreDNS is firmly rooted in open-source principles, fostering an active and collaborative community dedicated to its continuous improvement.
+
+CoreDNS's plugin-driven architecture empowers users to customize their DNS infrastructure to meet specific requirements, making it adaptable for a wide range of use cases, such as serving as a DNS server, forwarding DNS requests, or implementing advanced features. This adaptability positions CoreDNS as a critical component in cloud-native environments where dynamic service discovery and communication are essential.
+
+### Actors
+**CoreDNS Server**</br>
```

Comment:
Is this the only actor in CoreDNS? Are there any other components that work together to deliver all functionalities?

Reply:
Hi @ragashreeshekar,

The CoreDNS server relies on plugins for all functionality. These plugins are treated as gears in a system, rather than individual actors. There are no other actors in CoreDNS.